### PR TITLE
Spotlight Translator Help Prompt

### LINKED ISSUE #358: [Feature]: Add spotlight translator prompt for better UX
### Checklist

- [X] This is an issue for LiveTL, not Hyperchat. I will go to [the Hyperchat repo](https://github.com/LiveTL/HyperChat) to report an issue with Hyperchat.
- [X] This is an issue for the LiveTL web extension, not LiveTL iOS or LiveTL Android. I will go to the [iOS](https://github.com/LiveTL/ios) or [Android](https://github.com/LiveTL/android) repo to report issues for those platforms.
- [X] I have checked for a duplicate or similar issue. I made sure to check closed issues as well.
- [X] I am not reporting a bug. If I am, I will use the bug report issue template.
- [X] By submitting this issue, you agree to follow our [Code of Conduct](https://github.com/LiveTL/LiveTL/blob/develop/CODE_OF_CONDUCT.md)

### Is your feature request related to a problem? Please describe.

Many people do not know about the spotlight translator functionality as is shown by [the top reply to a reddit comment](https://reddit.com/r/Hololive/comments/r4dxff/_/hmg171f/?context=1).

### Describe the solution you'd like

If there are a certain amount of translators in the displayed messages, LiveTL should prompt the user if they want to spotlight a translator. This prompt should have an option to `never show again` as to not be annoying.

### Expected behavior

This is not a bug report lmao, we should remove this box.

### Describe alternatives you've considered

We could make reddit posts explaining that this feature exists.

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by r2dev2**
Before ready for review:

- [x] @KentoNishi can you replace the demo gif with a real gif explaining how to spotlight

Fixes #358

### CODE REVIEW

**Comment by KentoNishi** [comment]
that and styling, will do "soon"

**Comment by KentoNishi** [comment]
https://user-images.githubusercontent.com/38841491/147372938-0e161a28-7b97-4366-a381-e6f9458515c5.mp4

**Comment by KentoNishi** [comment]
WARNING: For the demo video to load, you need to fully reload the extension becuz there's a change to the WAR whitelist

**Comment by KentoNishi** [comment]
looks pretty good to me



https://user-images.githubusercontent.com/38841491/147373819-cc98093b-2293-4a49-adea-e002de5fa6ad.mp4

**Comment by KentoNishi** [comment]
OK this is ready for review

https://user-images.githubusercontent.com/38841491/147374270-2f382aaf-835a-4a7b-b3fd-c71b99df4188.mp4

**Review by KentoNishi — APPROVED** [review]

**Comment by KentoNishi** [comment]
small issue: it gets cut off if the window is too small, but i guess they can just resize the panel?

image

File: webpack.config.js [hunk 1]
```diff
@@ -213,7 +213,7 @@ module.exports = (env, options) => {
             }
           },
           {
-            from: 'src/img/blfavicon.ico',
+            from: 'src/img',
             to: 'img'
```
Comment (ChrRubin) [inline webpack.config.js:217]: Should change manifest to use icons from this folder and remove the old imports so they won't be duplicated.

File: src/ts/typings/types.d.ts [hunk 2]
```diff
@@ -1,3 +1,5 @@
+import { Writable } from 'svelte/store';
+
```
Comment (ChrRubin) [inline src/ts/typings/types.d.ts]: This will break the `declare` block, use inline imports instead.

File: src/ts/typings/types.d.ts [hunk 3]
```diff
@@ -19,4 +21,13 @@ declare namespace Ltl {
     hidden?: boolean;
     deleted?: boolean;
   }
+
+  interface FeaturePromptContent {
+    id: string;
+    prompt: string; // the prompt message for help
+    icon: string;
+    hasDismissed: Writable<boolean>;
+    neverShow: Writable<boolean>;
```
Comment (ChrRubin) [inline src/ts/typings/types.d.ts]: Refer above. `import('svelte/store').Writable` should do the trick.

File: src/js/featureprompt.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    const numAuthors = index(msgs).by('authorId').size;
+    if (numAuthors >= 3) {
+      pushPrompt(FeaturePrompt.SPOTLIGHT);
+    }
+  });
+
+  return () => {
+    unsubSpotlight();
+  };
+};
```
Comment (ChrRubin) [inline src/js/featureprompt.js]: Can't this be done in the svelte component script instead of being a separate script?

**Review by ChrRubin — CHANGES_REQUESTED** [review]

**Comment by KentoNishi** [comment]
all the changes requested are the parts that @r2dev2 so I'll leave them untouched. he's currently away on a trip until tuesday so we'll have to wait a bit

**Comment by KentoNishi** [comment]
https://user-images.githubusercontent.com/38841491/147418000-fb7d57c4-6502-47b9-bbb8-fc1f1b7375ac.mp4

**Comment by r2dev2** [comment]
@KentoNishi the close button is hidden on firefox
close button hidden

File: src/js/featureprompt.js (hunk 4 above)
Comment (r2dev2) [reply src/js/featureprompt.js]: Fixed in e378179a506cf60e94a1a5764354f21ea2fa0b4b

**Comment by KentoNishi** [comment]
@r2dev2 just resize the panel, it's not a browser issue

I'll probably make it so it doesn't overflow the screen

File: src/ts/typings/types.d.ts (hunk 3 above)
Comment (r2dev2) [reply src/ts/typings/types.d.ts]: Fixed in e378179a506cf60e94a1a5764354f21ea2fa0b4b

File: src/ts/typings/types.d.ts (hunk 2 above)
Comment (r2dev2) [reply src/ts/typings/types.d.ts]: Fixed in e378179a506cf60e94a1a5764354f21ea2fa0b4b

File: webpack.config.js (hunk 1 above)
Comment (r2dev2) [reply webpack.config.js:217]: Fixed in ca36467

**Comment by KentoNishi** [comment]
image

lolmao

**Review by ChrRubin — APPROVED** [review]
Except for the slight jank when the panel is too small, seems good to me.

**Comment by KentoNishi** [comment]
https://user-images.githubusercontent.com/38841491/147980091-bc22b788-2cbf-49b4-a61d-d7706f83f819.mp4


@r2dev2 merge if u approb

**Comment by KentoNishi** [comment]
https://user-images.githubusercontent.com/38841491/147990032-9aa5b8c0-b41d-4b62-b38e-6c5d8c8e9cf5.mp4

**Comment by r2dev2** [comment]
@KentoNishi just putting it here but after ur commits,

In open in LiveTL view, captions are partially hidden and the menu for plain/regex filters is borked.

caption sus

plain/regex filter menu bork

**Comment by KentoNishi** [comment]
@ChrRubin 
image

seems like the `alert` color is not taking effect in the prod build. i have no idea how css processing works for prod, can u help debug this?

**Comment by KentoNishi** [comment]
@r2dev2 final review plz

**Comment by r2dev2** [comment]
> @r2dev2 final review plz

I approve my own pull request.

**Pull request merged**